app/src/main/java/org/wikipedia/page/PageFragment.java

```diff
@@ -422,6 +422,7 @@ private void initWebViewListeners() {
             @Override
             public void onPageFinished(WebView view, String url) {
                 super.onPageFinished(view, url);
+                L.d("ttfp: " + activeTimer.getElapsedMillis());
```

Comment:
At this stage, the page is _nowhere near_ finished loading. Right after this line, we call the extremely heavy `JavaScriptActionHandler.setUp()` function, which rejiggers the entire DOM and causes multiple re-renders.  There is also the `setMargin()` call, which gets called even later, and causes the whole page to jump down below the lead image. We have to somehow stop the timer when _all of that_ is complete.